int32 key_up
int32 key_down
int32 key_left
int32 key_right
int32 key_left_shift
int32 key_left_ctrl